From 3d22322ccab57419f3afa06371055ec965e5844f Mon Sep 17 00:00:00 2001
From: Aswin Murugan <quic_aswinm@quicinc.com>
Date: Fri, 2 May 2025 20:45:55 +0530
Subject: [PATCH] PENDING: Create Mask for gpio input bit & read

In the current implementation the GPIO read is influenced by
GPIO output bit-1, ideally it should read the value of
input bit-0 alone, so created a mask to read bit-0.

Signed-off-by: Aswin Murugan <quic_aswinm@quicinc.com>
Upstream-Status: Pending
---
 drivers/gpio/msm_gpio.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpio/msm_gpio.c b/drivers/gpio/msm_gpio.c
index cea073b329..ffb189f988 100644
--- a/drivers/gpio/msm_gpio.c
+++ b/drivers/gpio/msm_gpio.c
@@ -189,7 +189,7 @@ static int msm_gpio_get_value(struct udevice *dev, unsigned int gpio)
 	if (qcom_is_special_pin(priv->pin_data, gpio))
 		return msm_gpio_get_value_special(priv, gpio);
 
-	return !!(readl(priv->base + GPIO_IN_OUT_REG(dev, gpio)) >> GPIO_IN);
+	return !!(readl(priv->base + GPIO_IN_OUT_REG(dev, gpio)) & BIT(GPIO_IN));
 }
 
 static int msm_gpio_get_function(struct udevice *dev, unsigned int gpio)
-- 
2.34.1

